compiler/resolution/functionResolution.cpp

```diff
@@ -9024,6 +9026,29 @@ static void resolveObviousGlobals() {
 }
 
 
+/************************************* | **************************************
+*                                                                             *
+*  Check for void fields as errors                                            *
+*                                                                             *
+************************************** | *************************************/
+
+static void checkNoVoidFields()
+{
+  for_alive_in_Vec(AggregateType, at, gAggregateTypes) {
+    // We exclude "no object" aggregates, to allow the Chapel internal 'ref' class
+    if(!at->symbol->hasFlag(FLAG_NO_OBJECT)) {
```

Comment:
The internal `ref` class has a definition similar to:
```
pragma "ref"
pragma "no default functions"
pragma "no object"
class _ref {
  var _val;
}
```
This causes `_val` to be implicitly declared as `void`. The exclusion of "no object" flagged objects prevents the field from being marked as an error.

Reply:
In `class _ref`, `var _val` is not implicitly declared as `void`. Rather, it is generic. Some code in the compiler is instantiating `_ref(void)` though. We could try to get it not to do that but this change is OK as well.